src/freenet/node/AnnounceSender.java

```diff
@@ -484,6 +484,15 @@ private void timedOut(PeerNode next) {
 	}
 
 	private void rnf(PeerNode next) {
+		synchronized(this) {
+			while(waitingForTransfers > 0) {
+				try {
+					wait();
+				} catch (InterruptedException e) {
+					// Ignore.
```

Comment:
Actually, I just copied this part of code from the complete()-method. Maybe it would be better to put this code into a private method that gets called at the beginning of rnf() and complete(). This method then could also handle exceptions raised during waiting.